### PR TITLE
Trim proxy port from System property

### PR DESCRIPTION
**Pull request by asodja**
A small fix that trims ports for http(s) proxy. It seems it's easy to have an empty space at the end and it's hard to identify the issue:
https://gradle-community.slack.com/archives/CAHSN3LDN/p1641377705434300

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

File: subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/JavaSystemPropertiesProxySettings.java [hunk 1]
```diff
@@ -58,11 +58,12 @@ private int initProxyPort(String proxyPortString) {
             return defaultPort;
         }
         try {
+            proxyPortString = proxyPortString.trim();
```
Comment (big-guy) [inline subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/JavaSystemPropertiesProxySettings.java:61]: How does this interact with the isBlank call above?

Comment (big-guy) [reply subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/JavaSystemPropertiesProxySettings.java:61]: And should we be trimming everything that we read from system properties?

Comment (asodja) [reply subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/JavaSystemPropertiesProxySettings.java:61]: `isBlank` checks that `proxyPortString` is not null or blank (just empty characters). So at least a null check would still be needed here if we remove it.

> And should we be trimming everything that we read from system properties?
I guess we could trim all system properties in this class. I am not sure just for username and password, does it make sense to have empty characters there?

Comment (asodja) [reply subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/JavaSystemPropertiesProxySettings.java]: @big-guy I changed the logic so we now trim all system properties. Could you take another look?

**Comment by asodja** [comment]
@bot-gradle test and merge

**Pull request merged**